### PR TITLE
N3 5 add authentication based on jwt access token

### PR DESCRIPTION
**Pull request by dmkofgl**

### CODE REVIEW

File: src/main/java/dl/news/portal/Application.java [hunk 1]
```diff
@@ -2,11 +2,18 @@
...
 
 @SpringBootApplication
 public class Application {
     public static void main(String[] args) {
         SpringApplication.run(Application.class, args);
     }
 
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder(5);
```
Comment (avinahradau) [inline src/main/java/dl/news/portal/Application.java]: magic number

File: src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public UserDetails loadUserByUsername(String s) throws UsernameNotFoundException {
+
+        UserDetails userDetails = userService.findUserByUsername(s)
+                .map(this::mapUser)
+                .orElseThrow(() -> new UsernameNotFoundException("No user found with username " + s));
+        return userDetails;
+    }
+
+    private UserDetails mapUser(User user) {
```
Comment (avinahradau) [inline src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java]: bad name

File: src/main/java/dl/news/portal/web/config/security/jwt/TokenProvider.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.Date;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class TokenProvider {
+
+    private static final String AUTHORITIES_KEY = "scopes";
+    private static final String SIGNING_KEY = "newsPortal";
+    private static final long ACCESS_TOKEN_VALIDITY_SECONDS = 5 * 60;
```
Comment (avinahradau) [inline src/main/java/dl/news/portal/web/config/security/jwt/TokenProvider.java]: dont use calculation here

File: src/main/java/dl/news/portal/web/config/security/jwt/AuthenticationController.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+public class AuthenticationController {
+
+    @Autowired
+    private AuthenticationManager authenticationManager;
+
+    @Autowired
+    private TokenProvider jwtTokenUtil;
+
+    @RequestMapping(value = "/generate", method = RequestMethod.POST)
+    public ResponseEntity getAuthToken(@RequestBody UserDto loginUser) throws AuthenticationException {
```
Comment (avinahradau) [inline src/main/java/dl/news/portal/web/config/security/jwt/AuthenticationController.java:26]: Add some tests

**Review by avinahradau — CHANGES_REQUESTED** [review]

File: src/main/java/dl/news/portal/web/config/security/jwt/TokenProvider.java (hunk 3 above)
Comment (dmkofgl) [reply src/main/java/dl/news/portal/web/config/security/jwt/TokenProvider.java]: Ok

File: src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java (hunk 2 above)
Comment (dmkofgl) [reply src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java]: Changed to createUserDetailsOfUser

File: src/main/java/dl/news/portal/web/config/security/jwt/AuthenticationController.java (hunk 4 above)
Comment (dmkofgl) [reply src/main/java/dl/news/portal/web/config/security/jwt/AuthenticationController.java:26]: Added 2 tests

File: src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java (hunk 2 above)
Comment (avinahradau) [reply src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java]: Still bad name :) change to mapUserToSOMETHING

Comment (dmkofgl) [reply src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java]: mapUserToUserDatails
 Is it right name?)

Comment (avinahradau) [reply src/main/java/dl/news/portal/domain/service/impl/UserDetailService.java]: yep, gj

**Review by avinahradau — APPROVED** [review]

**Pull request merged**